packages/bpk-component-autosuggestV2/docs/migrating-from-v1-to-v2.md

```diff
@@ -0,0 +1,101 @@
+# Migrating from `BpkAutosuggest` to `BpkAutosuggestV2`
+
+`BpkAutosuggest` has been completely rewritten to make it more accessible for screen reader users.
+
+## History and rationale
+
+The current base library is `react-autosuggest` which is no longer supported and is known to have a11y issues. We had two options: take over ownership of this library or write our own implementation.
+
+Banana has been using a custom implementation built using `downshift`'s `useCombobox` hook since November 2022 which gives screen reader users a better experience. The main improvement is the feedback when results are rendered.
+
+## Migrating
+
+* the component now accepts a ref instead of a ref callback
+
+**Before**
+
+```js
+autosuggestRef: typeof BpkAutosuggest;
+
+<BpkAutosuggest
+  ref={(bpkAutosuggest) => { this.autosuggestRef = bpkAutosuggest; }}>
+
+```
+**Replacement**
+
+
+```js
+autosuggestRef: React.RefObject<HTMLInputElement>;
+
+this.autosuggestRef = React.createRef<HTMLInputElement>();
+<BpkAutosuggest
+  ref={this.autosuggestRef}>
+
+```
+* aria strings are required:
+    * `getA11yResultsMessage` is a callback which returns the number of results being rendered. This should be used to pass translations for single, multiple and no results returned
+    * `ariaLabels` requires a `resultsList` string which will be appended to the `<ul>`
+
+* `renderInputComponent` is no longer a prop. The component renders a BpkInput component, any extension or modification of this can be done via the `inputProps` prop which is passed into the input or by extending or overriding the styling
```

Comment:
Its used within flights search controls. I'd looked at it as part of this draft pr to test the new component which is where I found we could do away with it - https://github.skyscannertools.net/web/flights-search-controls/pull/858

Reply:
Even if it is used, it should not have any effect as we set the `renderInputComponent` in BPK and you cannot override it - https://github.com/Skyscanner/backpack/blob/main/packages/bpk-component-autosuggest/src/BpkAutosuggest.js#L59

Reply:
I think given it is being used a migration guide away from using it is needed. Its not a prop on the new component so it can't be used in future but we don't want to block migrations by a consumer not knowing what to do with this now non existent prop